# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmGC.h
# Commit: 22c94980495f
# Full Hash: 22c94980495f7fe64f1bc554aa1d16da66b2d2cd
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 03:48:30
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 1: Reserve two slots after stack arguments for the future tls preservation. r=lth
#   
#   We are going to remove Frame::tls and support trampolines for indirect calls, so we need to get rid of using Frame::tls.
#   In this and the followup patches I will iteratively remove all dependencies of Frame::tls and remove it eventually.
#   
# ==============================================================================

diff -r 1a25820134c9 -r 22c94980495f js/src/wasm/WasmGC.h
--- a/js/src/wasm/WasmGC.h	Thu Oct 29 13:05:27 2020 +0000
+++ b/js/src/wasm/WasmGC.h	Fri Oct 23 18:17:36 2020 +0000
@@ -261,7 +261,7 @@
 // the complete native-ABI-level call signature.
 template <class T>
 static inline size_t StackArgAreaSizeUnaligned(const T& argTypes) {
-  ABIArgIter<const T> i(argTypes);
+  WasmABIArgIter<const T> i(argTypes);
   while (!i.done()) {
     i++;
   }
@@ -270,7 +270,7 @@
 
 static inline size_t StackArgAreaSizeUnaligned(
     const SymbolicAddressSignature& saSig) {
-  // ABIArgIter::ABIArgIter wants the items to be iterated over to be
+  // WasmABIArgIter::ABIArgIter wants the items to be iterated over to be
   // presented in some type that has methods length() and operator[].  So we
   // have to wrap up |saSig|'s array of types in this API-matching class.
   class MOZ_STACK_CLASS ItemsAndLength {